cli/cmd/contracts.go

```diff
@@ -94,6 +95,84 @@ var (
 		},
 	}
 
+	contractsStorageCmd = &cobra.Command{
+		Use:   "storage",
+		Short: "WebAssembly smart contracts storage operations",
+	}
+
+	contractsStorageDumpCmd = &cobra.Command{
+		Use:   "dump <instance-id>",
+		Short: "Dump instantiated public contract state in JSON",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := cliConfig.Global()
+			npa := common.GetNPASelection(cfg)
+			strInstanceID := args[0]
+
+			if npa.ParaTime == nil {
+				cobra.CheckErr("no paratimes configured")
+			}
+
+			instanceID, err := strconv.ParseUint(strInstanceID, 10, 64)
+			cobra.CheckErr(err)
+
+			ctx := context.Background()
+			conn, err := connection.Connect(ctx, npa.Network)
+			cobra.CheckErr(err)
+
+			res, err := conn.Runtime(npa.ParaTime).Contracts.InstanceRawStorage(ctx, client.RoundLatest, contracts.InstanceID(instanceID))
+			cobra.CheckErr(err)
+
+			storage := map[string]interface{}{}
+			for _, k := range res.Value {
+				var val interface{}
+				err = cbor.Unmarshal(k.Value, &val)
+				cobra.CheckErr(err)
```

Comment:
This should probably be made more robust, because the value encoding is up to the contract. You could try decoding as CBOR, but if it fails, you would output it as a Base64-encoded string in the dump.